### ISSUE TITLE
python3Packages.lektor fails to build: Could not find a version that satisfies the requirement mistune<2,>=0.7.0

### ISSUE THREAD

**Issue by exarkun** [NixOS/nixpkgs#191943]
### Steps To Reproduce
1. Against nixpkgs master, `nix-shell -p 'python3.withPackages(ps:[ps.lektor])'`

### Build log
```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pip-build-hook
Using pipBuildPhase
Using pipShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing pytest-check-hook
Using pytestCheckPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/i1nm25q0i23vvw14366b7m3yqm63rhxf-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tox.ini
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing pipBuildPhase
Creating a wheel...
WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permiss>
Processing /build/source
  Running command Preparing metadata (pyproject.toml)
  running dist_info
  creating /build/pip-modern-metadata-irvjhu5y/Lektor.egg-info
  writing /build/pip-modern-metadata-irvjhu5y/Lektor.egg-info/PKG-INFO
  writing dependency_links to /build/pip-modern-metadata-irvjhu5y/Lektor.egg-info/dependency_links.txt
  writing entry points to /build/pip-modern-metadata-irvjhu5y/Lektor.egg-info/entry_points.txt
  writing requirements to /build/pip-modern-metadata-irvjhu5y/Lektor.egg-info/requires.txt
  writing top-level names to /build/pip-modern-metadata-irvjhu5y/Lektor.egg-info/top_level.txt
  writing manifest file '/build/pip-modern-metadata-irvjhu5y/Lektor.egg-info/SOURCES.txt'
  reading manifest file '/build/pip-modern-metadata-irvjhu5y/Lektor.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  no previously-included directories found matching 'frontend'
  warning: no directories found matching 'lektor/admin/static'
  warning: no previously-included files matching '*.py[cdo]' found anywhere in distribution
  warning: no previously-included files matching '__pycache__' found anywhere in distribution
  warning: no previously-included files matching '*.so' found anywhere in distribution
  warning: no previously-included files matching '.DS_Store' found anywhere in distribution
  warning: no previously-included files matching '.gitkeep' found anywhere in distribution
  warning: no previously-included files matching '*~' found anywhere in distribution
  no previously-included directories found matching '.github'
  warning: no previously-included files found matching '.gitignore'
  warning: no previously-included files found matching '.git-blame-ignore-revs'
  warning: no previously-included files found matching '.codecov.yml'
  warning: no previously-included files found matching '.editorconfig'
  warning: no previously-included files found matching '.pre-commit-config.yaml'
  warning: no previously-included files found matching 'Makefile'
  warning: no previously-included files found matching 'pylintrc'
  no previously-included directories found matching 'example'
  adding license file 'LICENSE'
  writing manifest file '/build/pip-modern-metadata-irvjhu5y/Lektor.egg-info/SOURCES.txt'
  creating '/build/pip-modern-metadata-irvjhu5y/Lektor-0.0.0.dist-info'
  adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: Lektor
  Running command Building wheel for Lektor (pyproject.toml)
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/lektor
  copying lektor/__init__.py -> build/lib/lektor
  copying lektor/__main__.py -> build/lib/lektor
  copying lektor/assets.py -> build/lib/lektor
  copying lektor/build_programs.py -> build/lib/lektor
  copying lektor/builder.py -> build/lib/lektor
  copying lektor/buildfailures.py -> build/lib/lektor
  copying lektor/cli.py -> build/lib/lektor
  copying lektor/cli_utils.py -> build/lib/lektor
  copying lektor/constants.py -> build/lib/lektor
  copying lektor/context.py -> build/lib/lektor
  copying lektor/databags.py -> build/lib/lektor
  copying lektor/datamodel.py -> build/lib/lektor
  copying lektor/db.py -> build/lib/lektor
  copying lektor/devcli.py -> build/lib/lektor
  copying lektor/devserver.py -> build/lib/lektor
  copying lektor/editor.py -> build/lib/lektor
  copying lektor/exception.py -> build/lib/lektor
  copying lektor/filecontents.py -> build/lib/lektor
  copying lektor/i18n.py -> build/lib/lektor
  copying lektor/imagetools.py -> build/lib/lektor
  copying lektor/markdown.py -> build/lib/lektor
  copying lektor/metaformat.py -> build/lib/lektor
  copying lektor/packages.py -> build/lib/lektor
  copying lektor/pagination.py -> build/lib/lektor
  copying lektor/pluginsystem.py -> build/lib/lektor
  copying lektor/project.py -> build/lib/lektor
  copying lektor/publisher.py -> build/lib/lektor
  copying lektor/quickstart.py -> build/lib/lektor
  copying lektor/reporter.py -> build/lib/lektor
  copying lektor/sourceobj.py -> build/lib/lektor
  copying lektor/sourcesearch.py -> build/lib/lektor
  copying lektor/utils.py -> build/lib/lektor
  copying lektor/videotools.py -> build/lib/lektor
  copying lektor/watcher.py -> build/lib/lektor
  creating build/lib/lektor/admin
  copying lektor/admin/__init__.py -> build/lib/lektor/admin
  copying lektor/admin/utils.py -> build/lib/lektor/admin
  copying lektor/admin/webui.py -> build/lib/lektor/admin
  creating build/lib/lektor/environment
  copying lektor/environment/__init__.py -> build/lib/lektor/environment
  copying lektor/environment/config.py -> build/lib/lektor/environment
  copying lektor/environment/expressions.py -> build/lib/lektor/environment
  creating build/lib/lektor/types
  copying lektor/types/__init__.py -> build/lib/lektor/types
  copying lektor/types/base.py -> build/lib/lektor/types
  copying lektor/types/fake.py -> build/lib/lektor/types
  copying lektor/types/flow.py -> build/lib/lektor/types
  copying lektor/types/formats.py -> build/lib/lektor/types
  copying lektor/types/multi.py -> build/lib/lektor/types
  copying lektor/types/primitives.py -> build/lib/lektor/types
  copying lektor/types/special.py -> build/lib/lektor/types
  creating build/lib/lektor/admin/modules
  copying lektor/admin/modules/__init__.py -> build/lib/lektor/admin/modules
  copying lektor/admin/modules/api.py -> build/lib/lektor/admin/modules
  copying lektor/admin/modules/common.py -> build/lib/lektor/admin/modules
  copying lektor/admin/modules/dash.py -> build/lib/lektor/admin/modules
  copying lektor/admin/modules/serve.py -> build/lib/lektor/admin/modules
  running egg_info
  creating Lektor.egg-info
  writing Lektor.egg-info/PKG-INFO
  writing dependency_links to Lektor.egg-info/dependency_links.txt
  writing entry points to Lektor.egg-info/entry_points.txt
  writing requirements to Lektor.egg-info/requires.txt
  writing top-level names to Lektor.egg-info/top_level.txt
  writing manifest file 'Lektor.egg-info/SOURCES.txt'
  reading manifest file 'Lektor.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  no previously-included directories found matching 'frontend'
  warning: no directories found matching 'lektor/admin/static'
  warning: no previously-included files matching '*.py[cdo]' found anywhere in distribution
  warning: no previously-included files matching '__pycache__' found anywhere in distribution
  warning: no previously-included files matching '*.so' found anywhere in distribution
  warning: no previously-included files matching '.DS_Store' found anywhere in distribution
  warning: no previously-included files matching '.gitkeep' found anywhere in distribution
  warning: no previously-included files matching '*~' found anywhere in distribution
  no previously-included directories found matching '.github'
  warning: no previously-included files found matching '.gitignore'
  warning: no previously-included files found matching '.git-blame-ignore-revs'
  warning: no previously-included files found matching '.codecov.yml'
  warning: no previously-included files found matching '.editorconfig'
  warning: no previously-included files found matching '.pre-commit-config.yaml'
  warning: no previously-included files found matching 'Makefile'
  warning: no previously-included files found matching 'pylintrc'
  no previously-included directories found matching 'example'
  adding license file 'LICENSE'
  writing manifest file 'Lektor.egg-info/SOURCES.txt'
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/__init__.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/__main__.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/assets.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/build_programs.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/builder.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/buildfailures.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/cli.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/cli_utils.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/constants.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/context.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/databags.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/datamodel.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/db.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/devcli.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/devserver.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/editor.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/exception.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/filecontents.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/i18n.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/imagetools.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/markdown.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/metaformat.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/packages.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/pagination.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/pluginsystem.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/project.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/publisher.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/quickstart.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/reporter.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/sourceobj.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/sourcesearch.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/utils.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/videotools.py -> build/bdist.linux-x86_64/wheel/lektor
  copying build/lib/lektor/watcher.py -> build/bdist.linux-x86_64/wheel/lektor
  creating build/bdist.linux-x86_64/wheel/lektor/admin
  copying build/lib/lektor/admin/__init__.py -> build/bdist.linux-x86_64/wheel/lektor/admin
  copying build/lib/lektor/admin/utils.py -> build/bdist.linux-x86_64/wheel/lektor/admin
  copying build/lib/lektor/admin/webui.py -> build/bdist.linux-x86_64/wheel/lektor/admin
  creating build/bdist.linux-x86_64/wheel/lektor/admin/modules
  copying build/lib/lektor/admin/modules/__init__.py -> build/bdist.linux-x86_64/wheel/lektor/admin/modules
  copying build/lib/lektor/admin/modules/api.py -> build/bdist.linux-x86_64/wheel/lektor/admin/modules
  copying build/lib/lektor/admin/modules/common.py -> build/bdist.linux-x86_64/wheel/lektor/admin/modules
  copying build/lib/lektor/admin/modules/dash.py -> build/bdist.linux-x86_64/wheel/lektor/admin/modules
  copying build/lib/lektor/admin/modules/serve.py -> build/bdist.linux-x86_64/wheel/lektor/admin/modules
  creating build/bdist.linux-x86_64/wheel/lektor/environment
  copying build/lib/lektor/environment/__init__.py -> build/bdist.linux-x86_64/wheel/lektor/environment
  copying build/lib/lektor/environment/config.py -> build/bdist.linux-x86_64/wheel/lektor/environment
  copying build/lib/lektor/environment/expressions.py -> build/bdist.linux-x86_64/wheel/lektor/environment
  creating build/bdist.linux-x86_64/wheel/lektor/types
  copying build/lib/lektor/types/__init__.py -> build/bdist.linux-x86_64/wheel/lektor/types
  copying build/lib/lektor/types/base.py -> build/bdist.linux-x86_64/wheel/lektor/types
  copying build/lib/lektor/types/fake.py -> build/bdist.linux-x86_64/wheel/lektor/types
  copying build/lib/lektor/types/flow.py -> build/bdist.linux-x86_64/wheel/lektor/types
  copying build/lib/lektor/types/formats.py -> build/bdist.linux-x86_64/wheel/lektor/types
  copying build/lib/lektor/types/multi.py -> build/bdist.linux-x86_64/wheel/lektor/types
  copying build/lib/lektor/types/primitives.py -> build/bdist.linux-x86_64/wheel/lektor/types
  copying build/lib/lektor/types/special.py -> build/bdist.linux-x86_64/wheel/lektor/types
  running install_egg_info
  Copying Lektor.egg-info to build/bdist.linux-x86_64/wheel/Lektor-0.0.0-py3.10.egg-info
  running install_scripts
  adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
  creating build/bdist.linux-x86_64/wheel/Lektor-0.0.0.dist-info/WHEEL
  creating '/build/pip-wheel-aaig4u0s/tmpy83lfxjo/Lektor-0.0.0-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
  adding 'lektor/__init__.py'
  adding 'lektor/__main__.py'
  adding 'lektor/assets.py'
  adding 'lektor/build_programs.py'
  adding 'lektor/builder.py'
  adding 'lektor/buildfailures.py'
  adding 'lektor/cli.py'
  adding 'lektor/cli_utils.py'
  adding 'lektor/constants.py'
  adding 'lektor/context.py'
  adding 'lektor/databags.py'
  adding 'lektor/datamodel.py'
  adding 'lektor/db.py'
  adding 'lektor/devcli.py'
  adding 'lektor/devserver.py'
  adding 'lektor/editor.py'
  adding 'lektor/exception.py'
  adding 'lektor/filecontents.py'
  adding 'lektor/i18n.py'
  adding 'lektor/imagetools.py'
  adding 'lektor/markdown.py'
  adding 'lektor/metaformat.py'
  adding 'lektor/packages.py'
  adding 'lektor/pagination.py'
  adding 'lektor/pluginsystem.py'
  adding 'lektor/project.py'
  adding 'lektor/publisher.py'
  adding 'lektor/quickstart.py'
  adding 'lektor/reporter.py'
  adding 'lektor/sourceobj.py'
  adding 'lektor/sourcesearch.py'
  adding 'lektor/utils.py'
  adding 'lektor/videotools.py'
  adding 'lektor/watcher.py'
  adding 'lektor/admin/__init__.py'
  adding 'lektor/admin/utils.py'
  adding 'lektor/admin/webui.py'
  adding 'lektor/admin/modules/__init__.py'
  adding 'lektor/admin/modules/api.py'
  adding 'lektor/admin/modules/common.py'
  adding 'lektor/admin/modules/dash.py'
  adding 'lektor/admin/modules/serve.py'
  adding 'lektor/environment/__init__.py'
  adding 'lektor/environment/config.py'
  adding 'lektor/environment/expressions.py'
  adding 'lektor/types/__init__.py'
  adding 'lektor/types/base.py'
  adding 'lektor/types/fake.py'
  adding 'lektor/types/flow.py'
  adding 'lektor/types/formats.py'
  adding 'lektor/types/multi.py'
  adding 'lektor/types/primitives.py'
  adding 'lektor/types/special.py'
  adding 'Lektor-0.0.0.dist-info/LICENSE'
  adding 'Lektor-0.0.0.dist-info/METADATA'
  adding 'Lektor-0.0.0.dist-info/WHEEL'
  adding 'Lektor-0.0.0.dist-info/entry_points.txt'
  adding 'Lektor-0.0.0.dist-info/top_level.txt'
  adding 'Lektor-0.0.0.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Building wheel for Lektor (pyproject.toml) ... done
  Created wheel for Lektor: filename=Lektor-0.0.0-py3-none-any.whl size=129253 sha256=50a3cbd2f46bf3847841da6ebece422394323c6ee8189d5a9ec867d0cce2d7a6
  Stored in directory: /build/pip-ephem-wheel-cache-53qvgcyj/wheels/79/4b/9b/fc24941d75fdb959a71ee80195b2c89d6ae9389f9f0f80fb22
Successfully built Lektor
Finished creating a wheel...
Finished executing pipBuildPhase
@nix { "action": "setPhase", "phase": "installPhase" }
installing
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./Lektor-0.0.0-py3-none-any.whl
Requirement already satisfied: inifile>=0.4.1 in /nix/store/4bq0kr3dcm9wfmjqpz997nlk06zi6l3k-python3.10-inifile-0.4.1/lib/python3.10/site-packages (from Lektor==0.0.0) (0.4.1)
ERROR: Could not find a version that satisfies the requirement mistune<2,>=0.7.0 (from lektor) (from versions: none)
ERROR: No matching distribution found for mistune<2,>=0.7.0
```

### Notify maintainers

@costrouc 

### Metadata

> Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Dunno if it is relevant but ... 

```console
[exarkun@magnon:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.18.3, NixOS, 22.05 (Quokka), 22.05.915.9ff91ce2e4c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"home-manager-22.05, nixos-22.05.915.9ff91ce2e4c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by exarkun** [comment]
It looks like mistune 2.x compatibility was added in 9b8613da5e9b67924a3289553a709199f3d344b4 which included in Lektor v3.3.6.

**Comment by exarkun** [comment]
Err, nope, my mistake.  It is only in v3.4.0b1 and v3.4.0b2 right now.

**Comment by exarkun** [comment]
And 3.4.0 introduces a new dependency on (at least) the unpackaged `deprecated` package.

**Comment by fabaff** [comment]
> And 3.4.0 introduces a new dependency on (at least) the unpackaged deprecated package.

`deprecated` is [available](https://search.nixos.org/packages?channel=unstable&show=python310Packages.deprecated&from=0&size=50&sort=relevance&type=packages&query=deprecated).

**Comment by fabaff** [comment]
Was fixed as part of #194878.
